### ISSUE TITLE
imports on non public bits of code from libraries inside components

### ISSUE THREAD

**Issue by Waldbillig-Thomas** [grafana/grafana#40692]
**What happened**: I tried to fork the timeseries component to create a customized version of it but it contains a lot of 'bad' import, refering directly to some files instead of using the libraries public interfaces.

Some come from libraries, other refer directly to code relative to the project root.

In short it only works if you are in the original grafana repo, but could not possibly work when trying to do it in a panel generated from the @grafana/toolkit.

  Module not found: Error: Can't resolve '@grafana/ui/src/components/uPlot/utils' in '/workspace/packages/panels/bi-time-series-panel/src/plugins'
  Module not found: Error: Can't resolve '@grafana/ui/src/themes/mixins' in '/workspace/packages/panels/bi-time-series-panel/src/plugins/annotations'
  Module not found: Error: Can't resolve 'app/core/components/TagFilter/TagFilter' in '/workspace/packages/panels/pw-time-series-panel/src/plugins/annotations'
  Module not found: Error: Can't resolve 'app/features/alerting/state/alertDef' in '/workspace/packages/panels/bi-time-series-panel/src/plugins/annotations'
  Module not found: Error: Can't resolve 'app/features/annotations/api' in '/workspace/packages/panels/bi-time-series-panel/src/plugins/annotations'
  Module not found: Error: Can't resolve 'app/features/explore/utils/links' in '/workspace/packages/panels/bi-time-series-panel/src'
  
**What you expected to happen**:

being able to easily extend the original grafana components:
- no non-public imports
- no relative to root folder imports

**How to reproduce it (as minimally and precisely as possible)**:

- genrate plugin with the grafana-toolkit
- copy the code from a panel from the grafana repo (at a specific) tag to your plugin
- make sure that the dependency versions in your plugin match the tag of the the grafana version you copied it from
- add missing dependencies with the same version as in the grafana version you copied it from 
- yarn install && yarn build and get the same output I just posted above 

**Anything else we need to know?**:
tl;dr
could you also please just add a range to your dependencies instead of just putting 'latest'. that would make it maintaing/updating/automating multiple plugins so much easier.
have you considered using nx (https://nx.dev/)? it think it would really help avoiding issues like these and simplify a lot of the logic/scripts/automation inside your monorepo and would really help you to make your codebase consistent and structured

Thanks for the good work
Thomas

**Environment**:
- Grafana version: 8.2.1
- Data source type & version: n/a
- OS Grafana is installed on: Linux (but bot relevant in this case)
- User OS & Browser: Linux (but bot relevant in this case)
- Grafana plugins: any
- Others:
- 
![Screenshot_20211020_125642](https://user-images.githubusercontent.com/568657/138087595-532fd065-0c28-47c2-85d4-c8075030c1d0.png)

**Comment by mckn** [comment]
Closing this issue due to inactivity.
